tests/Unit/GuzzleFileFetcherTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Tuf\Tests\Unit;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+use Tuf\Client\GuzzleFileFetcher;
+use Tuf\Exception\DownloadSizeException;
+use Tuf\Exception\RepoFileNotFound;
+
+/**
+ * @coversDefaultClass \Tuf\Client\GuzzleFileFetcher
+ */
+class GuzzleFileFetcherTest extends TestCase
+{
+    /**
+     * Data provider for testFetchFile().
+     *
+     * @return array[]
+     *   Sets of arguments to pass to the test method.
+     */
+    public function dataProvider(): array
+    {
+        return [
+            [404, 256, RepoFileNotFound::class],
+            [403, 256, 'RuntimeException'],
```

Comment:
> why not use ::class in both case?

Because the fully qualified name of \RuntimeException is 'RuntimeException', and adding `::class` would be redundant, extra typing. I don't feel strongly about it, though, so I'll fix it if you want.

> Also we could just use new RuntimeException('the message') and then assert the class and messages match. Especially because we are using RuntimeException be good to check it is actually the one we are expecting.

The problem with this is that I don't know what error message will be generated by Guzzle for 403s, 500s, or other non-404 codes. And the file fetcher doesn't really need to care about that. How about, if we are expecting a \RuntimeException, I call `$this->expectExceptionCode($statusCode)` instead? That would confirm that the correct exception is being thrown.

Reply:
could we add just always call `$this->expectExceptionCode($statusCode)`.  everything else seems fine the way it is